cmd/prom-migrator/main.go

```diff
@@ -106,6 +108,18 @@ func main() {
 		os.Exit(0)
 	}
 
+	ms, err := parser.ParseExpr(conf.readerMetricsMatcher)
+	if err != nil {
+		log.Error("msg", "could not parse reader metrics matcher", "error", err)
+		os.Exit(2)
+	}
```

Comment:
Please shift the validation part in `func validateConf(conf *config) error`. We can keep this as the first check.

Reply:
Though about that but `parseFlags` fills config with simple types and `validateConf` doesn't change config fields.

So should I add raw validation (without setting some field of `config`) to `validateConf` and leave parsing string+type casting somewhere in `main`?

Reply:
I think we should not keep the parsing logic in main. We can do something like what `convertTimeStrFlagsToTs()` does in the `validateConf()`, and similarly, do the parsing in an external function that is called from `validateConf()`. WDYT?

Reply:
Done